src/App.tsx

```diff
@@ -1,61 +1,54 @@
-import React from 'react';
+import React, { useCallback, useState, useEffect } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Autocomplete } from './Autocomplete';
+import { debounce } from 'lodash';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState<string>('');
+  const [debounceQuery, setDebounceQuery] = useState<string>('');
+
+  useEffect(() => {
+    if (query !== '' && !selectedPerson) {
+      setSelectedPerson(null);
+    }
+  }, [query, selectedPerson])
 
-  return (
-    <div className="container">
-      <main className="section is-flex is-flex-direction-column">
-        <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
-        </h1>
 
-        <div className="dropdown is-active">
-          <div className="dropdown-trigger">
-            <input
-              type="text"
-              placeholder="Enter a part of the name"
-              className="input"
-              data-cy="search-input"
-            />
-          </div>
+  const filteredPeople = peopleFromServer.filter(person => person.name.toLowerCase().includes(debounceQuery.toLowerCase())
+  );
```

Comment:
Ensure that the filter is not called if the user enters spaces only. This is a requirement from the checklist: 'make sure that filter won't be called if user entered spaces only'.